src/MongoDB/Query.c

```diff
@@ -205,6 +205,32 @@ static bool php_phongo_query_init_readconcern(php_phongo_query_t *intern, zval *
 	return true;
 } /* }}} */
 
+/* Initialize the "maxAwaitTimeMS" option. Returns true on success; otherwise,
+ * false is returned and an exception is thrown.
+ *
+ * The "maxAwaitTimeMS" option is assigned to the cursor after query execution
+ * via mongoc_cursor_set_max_await_time_ms(). */
+static bool php_phongo_query_init_max_await_time_ms(php_phongo_query_t *intern, zval *options TSRMLS_DC) /* {{{ */
+{
+	if (php_array_existsc(options, "maxAwaitTimeMS")) {
```

Comment:
Existing code depends on all of our internal structs being calloc-ed, which is why I omitted this. Elsewhere, we have code segments that assume pointers are NULL if we haven't set them, which seems far more dangerous if the calloc behavior were ever to change.

I'm up for making a new ticket to explicitly assign all of these properties if you agree.